include/LieGroupControllers/impl/ControllerBase.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file ControllerBase.h
+ * @authors Giulio Romualdi
+ * @copyright This software may be modified and  distributed under the terms of
+ * the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef LIE_GROUP_CONTROLLERS_IMPL_CONTROLLER_BASE_H
+#define LIE_GROUP_CONTROLLERS_IMPL_CONTROLLER_BASE_H
+
+#include <LieGroupControllers/impl/traits.h>
+
+namespace LieGroupControllers
+{
+
+template <class _Derived> class ControllerBase
+{
+public:
+
+    using State = typename internal::traits<_Derived>::State;
+    using Vector = typename internal::traits<_Derived>::Vector;
+    using Gains = typename internal::traits<_Derived>::Gains;
+
+private:
+    _Derived& derived()
+    {
+        return *static_cast<_Derived*>(this);
+    }
+    const _Derived& derived() const
+    {
+        return *static_cast<const _Derived*>(this);
+    }
+
+public:
+    bool setState(const State& state);
+
+    bool setDesiredState(const State& state);
+
+    bool setFeedForward(const Vector& feedForward);
+
+    void setGains(const Gains& gains);
+
+    void computeControlLaw();
+
+    const Vector& getControl() const;
```

Comment:
Some little documentation on the usage or sequence of commands would be really helpful.